sdk/storage/src/clients/container_client.rs

```diff
@@ -0,0 +1,56 @@
+use crate::clients::StorageClient;
+use azure_core::errors::AzureError;
+use http::method::Method;
+use http::request::{Builder, Request};
+use std::sync::Arc;
+
+pub trait AsContainerClient<CN: Into<String>> {
+    fn as_container_client(&self, container_name: CN) -> Arc<Box<ContainerClient>>;
+}
+
+impl<CN: Into<String>> AsContainerClient<CN> for Arc<Box<StorageClient>> {
+    fn as_container_client(&self, container_name: CN) -> Arc<Box<ContainerClient>> {
```

Comment:
There's no reason to wrap `ContainerClient` in a `Box`. `Arc`s are already heap allocated pointers.

Reply:
Ah well... 

![image](https://user-images.githubusercontent.com/431595/102233565-b5e9be80-3ef0-11eb-989d-d9f013a9ad6c.png)